indexer/editable_map_object.cpp

```diff
@@ -312,6 +312,15 @@ bool EditableMapObject::ValidateEmail(string const & email)
   if (email.empty())
     return true;
 
+  if (strings::IsASCIIString(email))
+    return regex_match(email, regex("^[^@\\s]+@[\\w-]+(\\.[\\w-]+)+$"));
```

Comment:
I guess you need to support underscores too.
Also, please use raw string literals instead of native C/C++ literals - they'll allow you to get rid of duplicating slashes.